Add synced tabs persistence (#2087)
withastro/starlight: .changeset/silver-houses-lick.md

---
'@astrojs/starlight': minor
---

> Adds persistence to synced `<Tabs>` so that user's choices are reflected throughout the entire website.

Reviewer: Suggesting a small tweak here:

```suggestion
Adds persistence to synced `<Tabs>` so that a user's choices are reflected across page navigations.
```
